src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java

```diff
@@ -871,7 +894,10 @@ private static void addPlaces(Leg leg, State[] states, Edge[] edges, boolean sho
 
         leg.from = makePlace(states[0], firstVertex, edges[0], firstStop, tripTimes, requestedLocale);
         leg.from.arrival = null;
-        leg.to = makePlace(states[states.length - 1], lastVertex, null, lastStop, tripTimes, requestedLocale);
+        int lastTransitStateIndex = states[states.length - 1].backEdge instanceof TransitBoardAlight
+            ? states.length - 2
+            : states.length - 1;
+        leg.to = makePlace(states[lastTransitStateIndex], lastVertex, null, lastStop, tripTimes, requestedLocale);
```

Comment:
I would refactor the three instances of computing `lastTransitStateIndex` getting the last transit state into 
```java
State lastTransitState = getLastTransitState();
```
and replace all the instances of `states[lastTransitStateIndex]` with that variable.